### PR TITLE
[FFmpeg] drop use of deprecated methods

### PR DESCRIPTION
**Pull request by lrusak**
Not sure if this is exactly correct. Maybe @FernetMenta can school me :stuck_out_tongue_winking_eye: 

Could use testing.

### CODE REVIEW

**Comment by FernetMenta** [comment]
Incorrect usage of the API. avcodec_send_frame / avcodec_receive_packet can return EAGAIN, which has to be considered as valid return values. Clients have to act accordingly.

**Comment by lrusak** [comment]
updated according to https://ffmpeg.org/doxygen/trunk/encode_audio_8c-example.html

File: xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp [hunk 1]
```diff
@@ -260,37 +260,38 @@ int CAEEncoderFFmpeg::Encode(uint8_t *in, int in_size, uint8_t *out, int out_siz
...
   }
 
   /* initialize the output packet */
   av_init_packet(&m_Pkt);
   m_Pkt.size = out_size;
   m_Pkt.data = out;
 
-  ret = avcodec_receive_packet(m_CodecCtx, &m_Pkt);
-  if (ret < 0)
+  while (ret >= 0)
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp]: this loop is obviously wrong. you overwrite and trash already received packets.

**Comment by FernetMenta** [comment]
btw: you can't change the API from a synchronous one to an asynchronous one without adapting the client side. client has to cope with server wanting no data and delivering no data. currently this is not the case.
it's not just changing the API calls, that's the reason it has not been done so far.

**Comment by phunkyfish** [comment]
@lrusak do you want to progress this, close it or backburner?

**Comment by lrusak** [comment]
This would be a good candidate for someone looking for something to work on.

**Pull request closed without merging**